tests/lib/tools/setup_nested_hybrid_system.sh

```diff
@@ -0,0 +1,303 @@
+    # and we don't need the classic image anymore
+    # TODO: uncomment
+    # rm -f  "$NESTED_IMAGES_DIR/$(nested_get_image_name classic)"
+    # TODO: this prevents "nested_prepare_ssh" inside nested_start_core_vm
+    #       from running, we already have a user so this is not needed
+    local image_name
+    image_name="$(nested_get_image_name core)"
+    touch "${NESTED_IMAGES_DIR}/${image_name}.configured"
+    tests.nested create-vm core --extra-param "${NESTED_PARAM_EXTRA}"
+
+    # bind mount new seed
+    remote.exec "sudo mount -o bind /var/lib/snapd/install-seed /var/lib/snapd/seed"
+    # push and install muinstaller
+    remote.push "${muinstaller_snap}"
+    remote.exec "sudo snap install --classic --dangerous $(basename "${muinstaller_snap}")"
+
+    # run installation
+    local install_disk
+    install_disk=$(remote.exec "readlink -f /dev/disk/by-id/virtio-target")
+    remote.exec "sudo muinstaller classic \
+        ${install_disk} /snap/muinstaller/current/bin/mk-classic-rootfs.sh"
+
+    remote.exec "sudo sync"
+
+    # Stop and remove the classic vm now that the attached disk (${fake_disk})
+    # contains a just installed UC image.
+    tests.nested vm remove
+    sync
+
+    # HACK: rename to "core" image because we need to boot with OVMF
+    # we really should fix this so that classic and core VMs are more similar
+    mv "${fake_disk}" "${NESTED_IMAGES_DIR}/${image_name}"
+
+    # Change seed part label to capitals so we cover that use case
+    image_path="${NESTED_IMAGES_DIR}/${image_name}"
+    kpartx -asv "${image_path}"
+    fatlabel /dev/disk/by-label/ubuntu-seed UBUNTU-SEED
+    kpartx -d "${image_path}"
+
+    if [ -n "${store_dir}" ]; then
+        # if we had a store setup, then we should bring it back up
+        "${TESTSTOOLS}/store-state" setup-fake-store "${store_dir}"
+    fi
+
+    # Start installed image
+    tests.nested create-vm core --tpm-no-restart
+}
+
+main() {
+    local model_assertion=""
+    local store_dir=""
+    local gadget_snap=""
+    local gadget_assertion=""
+    local kernel_snap=""
+    local kernel_assertion=""
+    local label="classic"
+    local fake_disk=""
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --model)
+                model_assertion="${2}"
+                shift 2
+                ;;
+            --store-dir)
+                store_dir="${2}"
+                shift 2
+                ;;
+            --gadget)
+                gadget_snap="${2}"
+                shift 2
+                ;;
+            --gadget-assertion)
+                gadget_assertion="${2}"
+                shift 2
+                ;;
+            --kernel)
+                kernel_snap="${2}"
+                shift 2
+                ;;
+            --kernel-assertion)
+                kernel_assertion="${2}"
+                shift 2
+                ;;
+            --label)
+                label="${2}"
+                shift 2
+                ;;
+            --fake-disk)
```

Comment:
nitpick: in retrospective, fake-disk is not a great name, maybe this should be just "disk".